### PR TITLE
Feature/process death logic

### PR DESCRIPTION
**Pull request by ArtemisSoftware**

### CODE REVIEW

**Review by MKiperszmid — APPROVED** [review]

File: app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/eventdetail/EventDetailViewModel.kt [hunk 1]
```diff
@@ -62,13 +63,24 @@ class EventDetailViewModel @Inject constructor(
...
-    private val _state = MutableStateFlow(EventDetailState())
+    private val _state = MutableStateFlow(getState() ?: EventDetailState())
     val state: StateFlow<EventDetailState> = _state.asStateFlow()
 
     init {
         loadDetail()
     }
 
+    private fun updateState() {
+        savedStateHandle["state"] = _state.value
```
[the remark this answers is not in the document]
Comment (philipplackner) [reply app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/eventdetail/EventDetailViewModel.kt:74]: `const` is only for primitive fields

[the remark this answers is not in the document]
Comment (philipplackner) [reply app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/eventdetail/EventDetailViewModel.kt:74]: You could combine this with StateFlow's `update` function:
```
private fun updateState(update: (EventState) -> EventState) {
    savedStateHandle["state"] =  _state.updateAndGet { update(it) }
}
```

File: app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/eventdetail/EventDetailViewModel.kt [hunk 2]
```diff
@@ -214,6 +187,7 @@ class EventDetailViewModel @Inject constructor(
                 attendees = it.attendees + attendee,
             )
         }
+        updateState()
```
[the remark this answers is not in the document]
Comment (philipplackner) [reply app/src/main/java/com/artemissoftware/tasky/agenda/presentation/detail/eventdetail/EventDetailViewModel.kt:190]: With my suggestion, he can just use the `updateState` function and it'll be applied after every update call. The problem with just putting it at the end of onEvent is that it's only updated after each function runs and not immediately after updating the state which might lead to old values if you refer to `state.value`

**Review by philipplackner — APPROVED** [review]

File: app/src/main/java/com/artemissoftware/tasky/MainActivity.kt [hunk 3]
```diff
@@ -68,14 +69,10 @@ private fun MainScreen(viewModel: MainViewModel, navController: NavHostControlle
             startRoute = it,
             navController = navController,
         )
-    } ?: run {
-        DestinationsNavHost(
-            navGraph = NavGraphs.root,
-            startRoute = LoginScreenDestination,
-            navController = navController,
-        )
+        navController
```
[the remark this answers is not in the document]
Comment (ArtemisSoftware) [reply app/src/main/java/com/artemissoftware/tasky/MainActivity.kt:72]: After process death I need to add this navController or else the app will not navigate to the right screen

**Pull request merged**